### ISSUE TITLE
Add fallback for product meta lookup table

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/woocommerce#23960]
**Is your feature request related to a problem? Please describe.**
If the product meta lookup table fails to populate, some features no longer work, e.g. sorting by SKU, price, etc. Would it be possible to detect this and fall back to previous, less optimal queries in that case?

**Describe the solution you'd like**
Have a working fallback when possible.

**Describe alternatives you've considered**
- Improving reliability of AS... it's being discussed as well
- Have a more reliable notice for users when we detect a lot of pending jobs

**Additional context**
p7bje6-1mH-p2

**Comment by kloon** [comment]
There are fallbacks in some places but only while the upgrade routine is running. I think the reasoning behind us not having something in place everywhere and for outdated DB versions is because it would be too expensive to work out if it needs to fall back or not based on data.

That said we can maybe be more aggressive around our database updates, if they are so beneficial to a release why are we treating it an optional thing allowing a store to continue operating without doing it? Perhaps we need to be more clear around DB updates and the consequences of not running it, notices easily get lost among the already flooded screen space, perhaps an approach like how WP does this would be better?

![wp-db-update](https://user-images.githubusercontent.com/271630/59843401-c9968200-9358-11e9-9550-b6e678c92241.jpg)

**Comment by peterfabian** [comment]
I agree, db update is important and we should try to improve the reliability and communication around the upgrade routine, as getting scheduled actions running will be more and more important going forward.

Though I still think we can explore if we can have a fallback solution, just as another line of defence.

Maybe I'm imagining it too simplified, but e.g. if the number of products in the lookup table is lower than the number of products in regular products table, then we can't rely on the lookup table and we can use the fallback. (We could even cache this if needed and refresh it once a day?) I know it's not 100% true that if counts are the same, the data in the lookup table is correct, but it gives us at least some hint that AS probably runs. I'm quite sure other people can come up with better solutions, but I'd say reliability > speed, so if we're sure lookup table is empty, why try using it? Or do you think it does not make sense to have any fallback?

**Comment by DavidAnderson684** [comment]
Why does WC force the user to login and take manual action to update their database? Why does it not simply initiate the database updates straight away?

Users who update their plugins via an external dashboard (e.g. ManageWP) won't even know that that there's something to do.

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It met the criteria outlined there. If this still affects a current version of WooCommerce, please follow the steps in that post. A fresh report with current reproduction steps is the fastest path to triage.
